### ISSUE TITLE
Why is negative missing when calculating migration in ad_dif_mig_conductance model?

### ISSUE THREAD

**Issue by mkaguer** [PMEAL/OpenPNM#1723]
This is a question I had that I would like to open up for discussion. When you go to [line 153](https://github.com/PMEAL/OpenPNM/blob/ceeace0c753a492dc61f6a4749cd81d3d4a7089e/openpnm/models/physics/ad_dif_mig_conductance.py#L153) of ad_dif_mig_conductance model, migration is calculated but it is missing the negative sign out in front. The convention it uses to calculate mij is opposite to the convention used to calculate qij. Shouldn't these conventions be consistent with one another? 

Note that the convention is whether openpnm uses qij = Gij(ci - cj) or qij = Gij(cj - ci). I believe openpnm uses the first convention where j is the neighbour pore and i is the central pore but not certain. Anyone know for sure?

The conductance model goes on to calculate adv_mig which is qij - mij [(line 160)](https://github.com/PMEAL/OpenPNM/blob/ceeace0c753a492dc61f6a4749cd81d3d4a7089e/openpnm/models/physics/ad_dif_mig_conductance.py#L160). I believe the negative sign here corrects for the wrong convention used when calculating mij. However, I don't think that was the intent. I believe the intent of @magnaou when he wrote this model was to follow the equation as written about in his nernst planck paper. I included below a photo of equation 18 in the paper where he uses qij - mij. However, look at the equation and let me know if you think it could be wrong. I think it should be qij + mij. I believe this would be consistent with upwind scheme, no?

Bottom line: code is working correctly it's just using a bad equation that was fixed by using the wrong convention when calculating migration. Two wrongs made it right??

![eq18](https://user-images.githubusercontent.com/57453394/97355716-ab566580-186d-11eb-85f0-3108f78756b7.png)

**Comment by Ni2M** [comment]
About your first question, in the steady state case that I am working I realized the convention is qij = Gij(ci - cj), as in a simple case of fickian diffusion we have positive sign for the diagonal elements and negative sign for the off-diagonal one (the neighbouring terms). Still that would depend on what is the source term's sign. I think you can still change the convention, if you still use their correct sign for the source terms in the new convention (it's like you're multiplying the entire equation by -1 so it shouldn't change the result).

**Comment by mkaguer** [comment]
Hi @Ni2M. Thanks for giving your input. So you are confirming that the openpnm convention is qij = Gij(ci - cj)? The convention then should be the same for qij and mij but it is not. However, I think this is corrected later in the code when it uses qij - mij instead of qij + mij. I think the formula (eqn 18) above from the nernst Planck paper has the wrong sign. Does anyone have an opinion on that?

**Comment by ma-sadeghi** [comment]
It's probably best to ask @magnaou.

**Comment by mkaguer** [comment]
@ma-sadeghi I asked @magnaou before I created this issue. He said the negative is missing because of an OpenPNM convention and suggested I talk to @jgostick for more on that. However, Jeff and I agreed that the convention should be the same for migration and flow. Maybe @magnaou  can explain why it is qij - mij instead of qij + mij?

**Comment by ma-sadeghi** [comment]
@mkaguer Any update on this?

**Comment by mkaguer** [comment]
Hi @ma-sadeghi. The code is correct. I remember that two wrongs made it right. We can correct both errors but technically code works fine. I have to look back to see exactly where the two errors were.

**Comment by mkaguer** [comment]
I say we close this issue. It works.
